src/Nox.Generator/Application/Factories/EntityFactory.template.cs

```diff
@@ -53,11 +53,19 @@ public abstract class {{className}}Base: IEntityFactory<{{entity.Name}},{{entity
     {
         var entity = new {{codeGeneratorState.DomainNameSpace}}.{{entity.Name}}();
         {{- for key in entity.Keys }}
-            {{- if key.Type == "Nuid" || key.Type == "AutoNumber" || key.Type == "DatabaseGuid" -}}
+            {{- if key.Type == "Nuid" || key.Type == "AutoNumber" || key.Type == "Guid" -}}
```

Comment:
We should remove this check if the type is "Guid"

Reply:
If we remove this line there will be double line for generating Id, so I belive this check should stay here. I check whether id is null or not and call Create method from entity.
What do you think?

Reply:
I don't think we need 2 "for" loops for the Keys, it would be much easier if we handled all the cases in only 1 (first one) "for" loop, but it is not wrong even like this.
@rochar I would wait for your feedback here, please.

Reply:
is it not more simple a single for with if else?:
```
 {{- if key.Type == "Nuid" }}
        entity.Ensure{{key.Name}}();
{{- else if key.Type == "Guid" }}
         ....
{{- else if key.Type == "AutoNumber" }}
        {{ continue; -}}
{{ else }}
         entity.{{key.Name}} = {{ entity.Name }}.Create{{key.Name}}(createDto.{{key.Name}});
{{- end }}
```

Reply:
Since Nuid needs to be end of the method, it stays as is. 
Could you please check the latest implementation.